### ISSUE TITLE
qa: secrets are printed in Buildkite UI

### ISSUE THREAD

**Issue by bobheadxi** [sourcegraph/sourcegraph#16647]
As noted in https://github.com/sourcegraph/sourcegraph/issues/16207. For example https://buildkite.com/sourcegraph/qa/builds/957#dd812dd8-a912-4e54-b6cc-10b137404c02 => click any step and go to "Environment" tab

Buildkite logs require authentication to see (the above link doesn't work in incognito for example), but this is 

Opening separate issue for this for discussion/ideas, not sure where else these secrets should go (and/or if this is a concern)

**Comment by bobheadxi** [comment]
@davejrt assigned you for now since I believe you set this up, but happy to help out!

**Comment by davejrt** [comment]
@pecigonzalo I know you've used vault in the past. 

Was this the plugin used your, or something similar with AWS secrets manager?

https://github.com/mikeknox/vault-secrets-buildkite-plugin

**Comment by pecigonzalo** [comment]
No, we had our own using AWS SSM + Chamber. Chamber can export secrets in `dotenv` format, which you can directly source in a shell like `source <(chamber export SECRET -f dotenv)` (remember to unset `-x` or other things).

I can show you how I did it/where/why etc.

That plugin seems really stale tbh.

PS: I would remove this from the project, otherwise we keep increasing its scope.

**Comment by bobheadxi** [comment]
Removed from project 👍

**Comment by bobheadxi** [comment]
@davejrt Did we end up following on this?

**Comment by jhchabran** [comment]
@davejrt Just for the context, we are looking at this in order to check if we can give access to contractors to buildkite, on their own builds, without leaking secret. 

More context on urgency here https://sourcegraph.slack.com/archives/CJX299FGE/p1632217974252500

**Comment by valerybugakov** [comment]
It's still super relevant because, at the moment, the only way to share Buildkite errors with the contractor is through the comment on the PR. This approach doesn't scale. Happy to help with the solution if we have something in mind already!

cc @davejrt, @jhchabran, @bobheadxi

**Comment by jhchabran** [comment]
After reviewing how we deal with our secrets, I have not seen any blocker to make the `sourcegraph` pipeline public:

Buildkite [3.31](https://github.com/buildkite/agent/releases/tag/v3.31.0) enables secrets redaction by default and will redact the values of the following environment variables: 
-  `*_PASSWORD,*_SECRET,*_TOKEN,*_ACCESS_KEY,*_SECRET_KEY` 
- we have some variables named `*_CREDENTIALS`, they [are being added to the redacted vars](https://github.com/sourcegraph/infrastructure/pull/2797). 

To be perfectly clear, it means that if present in the logs, those variables values appear as `[REDACTED]`. 

- Our [list of secrets injected in Buildkite agents](https://github.com/sourcegraph/infrastructure/blob/main/buildkite/kubernetes/buildkite-agent/buildkite-agent.Deployment.yaml#L32-L220) will not redact notably:
  - `AZURE_DEVOPS_USERNAME`, `AWS_ACCESS_KEY_ID`. 
  - Those are not meant to be private, but some organisations like to consider them as such. 
  - :arrow_right: @sourcegraph/security could you confirm that we are ok with them being public? 
  
## QA 

I ran a build that explicitly try to print a secret here and the `PERCY_TOKEN` is being redacted. 

<img width="1038" alt="CleanShot 2021-09-30 at 09 56 19@2x" src="https://user-images.githubusercontent.com/10151/135411462-a1ac79ac-9816-4098-b56e-76761acd6e07.png">

## TODO

- Waiting on https://github.com/sourcegraph/infrastructure/pull/2797 to be merged before
- QA again with `_CREDENTIALS`

**Comment by andreeleuterio** [comment]
@jhchabran all good with Security. We have other projects to harden this further in the future but nothing atm. Thanks for looping us in!

**Comment by jhchabran** [comment]
## QA 

🔴 Failed 

<img width="1336" alt="CleanShot 2021-09-30 at 18 51 38@2x" src="https://user-images.githubusercontent.com/10151/135497938-9ed6ab32-9253-452e-bbcc-e1b176b14451.png">

--- 

Will try again tomorrow.

**Comment by jhchabran** [comment]
## QA

Ok, it works now. It's interesting to see that env defined within the pipeline steps are not redacted. I'll document all of that tomorrow! 

<img width="1242" alt="CleanShot 2021-09-30 at 22 06 40@2x" src="https://user-images.githubusercontent.com/10151/135523255-341412bd-e08e-475d-8bb7-dc05c2068e4e.png">

**Comment by jhchabran** [comment]
Update: We cannot remove previous builds, which includes a bunch of tokens, whom I don't know if they are still in use or not. That sounds like a tedious task. I think that the best course of action is to 

1. clone the `Sourcegraph` pipeline
2. rename the old one to `archived-Sourcegraph` so we have an archive (we can decide to drop it later on)
3. rename the new one to `Sourcegraph` 

This aligns with the potential changes of RFC 484 on the pipelines.

**Comment by jhchabran** [comment]
I'm closing this, we're using a freshly cloned branch with no history for the contractors.
